app/controllers/photos_controller.rb

```diff
@@ -0,0 +1,65 @@
+class PhotosController < ApplicationController
+
+before_action :authenticate_user!
+before_action :set_post, only: [:show, :edit, :update, :destroy]
+before_action :owned_photo, only: [:edit, :update, :destroy]
+
+  def index
+    @photos = Photo.all
+  end
+
+  def new
+    @photo = current_user.photos.build
+  end
+
+  def create
+     @photo = current_user.photos.build(photo_params)
+
+      if @photo.save
+        flash[:success] = "Your post has been created!"
+        redirect_to photos_path
+      else
+        flash.now[:alert] = "Your new post couldn't be created! Please check the form."
+        render :new
+  end
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+    if @photo.update(photo_params)
+      flash[:success] = "Post Updated."
+      redirect_to photos_path
+  else
+    flash.now[:alert] = "Update failed. Please check the form."
+    render :edit
+  end
+  end
```

Comment:
again - controller looks good, indentation would fail you, if this was a tech test